beetsplug/discogs.py

```diff
@@ -397,14 +398,28 @@ def get_tracks(self, tracklist):
         tracks = []
         index_tracks = {}
         index = 0
+        # Distinct works and intra-work divisions, as defined by index tracks.
+        divisions, next_divisions = [], []
         for track in clean_tracklist:
             # Only real tracks have `position`. Otherwise, it's an index track.
             if track['position']:
                 index += 1
-                track_info = self.get_track_info(track, index)
+                if next_divisions:
```

Comment:
Seems like this `if` may be unnecessary? The "shift" to `divisions` will work the same even if `next_divisions` is empty.

Reply:
That's true. I thought the conditional made for easier reading, since it makes clear that the shift only happens on the boundary between an index track and a real track. But I'm happy to remove it if you'd prefer.

Reply:
Aha, I get it. Sure, keeping it for clarity makes sense!